src/utility/ContainerUtil.h

```diff
@@ -172,6 +172,119 @@ auto clone(const std::vector<A>& xs) {
     return ys;
 }
 
+// -------------------------------------------------------------
+//                            Iterators
+// -------------------------------------------------------------
+
+/**
+ * A wrapper for an iterator obtaining pointers of a certain type,
+ * dereferencing values before forwarding them to the consumer.
+ *
+ * @tparam Iter ... the type of wrapped iterator
+ * @tparam T    ... the value to be accessed by the resulting iterator
+ */
+template <typename Iter, typename T = typename std::remove_pointer<typename Iter::value_type>::type>
+struct IterDerefWrapper : public std::iterator<std::forward_iterator_tag, T> {
+    /* The nested iterator. */
+    Iter iter;
+
+public:
+    // some constructores
```

Comment:
I know you moved this over from the old header file, but `constructores` -> `constructors`

Reply:
Also fixed the same typo in CompiledTuple.h